WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift

```diff
@@ -387,6 +387,7 @@ final class WebAddressWizardContent: UIViewController {
             return
         }
 
+        table.headerView(forSection: 0)?.isHidden = true
```

Comment:
If a user navigates Back one screen in the flow, then returns, you may find that the header is still not visible without resetting this attribute.

Another way to achieve the desired result : consider modifying `.tableView(_:titleForHeaderInSection:)` to return `nil` in this case (i.e., if/when the `searchTerm` is specified). This would likely require some further coordination between `WebAddressWizardContent` & `WebAddressTableViewProvider`.